src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+
+    public sealed partial class ECDsaCng : ECDsa
+    {
+        private static CngKey ImportKeyBlob(byte[] ecBlob, ECNamedCurve curve, bool includePrivate)
+        {
+            CngKeyBlobFormat blobFormat = includePrivate ? s_eccPrivateBlob : s_eccPublicBlob;
+
+            CngKey newKey = CngKey.Import(ecBlob, curve, blobFormat);
+            newKey.ExportPolicy |= CngExportPolicies.AllowPlaintextExport;
+
+            return newKey;
+        }
+
+        private static CngKey ImportFullKeyBlob(byte[] ecBlob, bool includePrivate)
+        {
+            CngKeyBlobFormat blobFormat = includePrivate ? s_eccFullPrivateBlob : s_eccFullPublicBlob;
+
+            CngKey newKey = CngKey.Import(ecBlob, blobFormat);
+            newKey.ExportPolicy |= CngExportPolicies.AllowPlaintextExport;
+
+            return newKey;
+        }
+
+        private static byte[] ExportKeyBlob(CngKey key, bool includePrivate)
+        {
+            return key.Export(includePrivate ? s_eccPrivateBlob : s_eccPublicBlob);
+        }
+
+        private static byte[] ExportFullKeyBlob(CngKey key, bool includePrivate)
+        {
+            return key.Export(includePrivate ? s_eccFullPrivateBlob : s_eccFullPublicBlob);
+        }
+
+        // CngKeyBlob formats for ECC key blobs
+        private static readonly CngKeyBlobFormat s_eccPrivateBlob = new CngKeyBlobFormat(Interop.BCrypt.KeyBlobType.BCRYPT_ECCPRIVATE_BLOB);
+        private static readonly CngKeyBlobFormat s_eccPublicBlob = new CngKeyBlobFormat(Interop.BCrypt.KeyBlobType.BCRYPT_ECCPUBLIC_BLOB);
+        private static readonly CngKeyBlobFormat s_eccFullPrivateBlob = new CngKeyBlobFormat(Interop.BCrypt.KeyBlobType.BCRYPT_ECCFULLPRIVATE_BLOB);
+        private static readonly CngKeyBlobFormat s_eccFullPublicBlob = new CngKeyBlobFormat(Interop.BCrypt.KeyBlobType.BCRYPT_ECCFULLPUBLIC_BLOB);
+
+        #region Import Parameters
+
+        /// <summary>
+        ///         ImportParameters will replace the existing key that ECDsaCng is working with by creating a
+        ///         new CngKey. If the parameters structure contains only Q,
+        ///         then only a public key will be imported. If the parameters also contains D, 
+        ///         then a full key pair will be imported. The Curve value is required to specify the what
+        ///         other values are imported.
+        /// </summary>
+        /// <exception cref="CryptographicException">
+        ///     if <paramref name="parameters" /> is not a valid ECDsa key or is missing values.
+        /// </exception>
+        public override void ImportParameters(ECParameters parameters)
+        {
+            int cbKey;
+            if (!ValidateKey(parameters, out cbKey))
+                throw new CryptographicException(SR.Cryptography_InvalidECKeyParameters);
+
+            bool includePrivate;
+            if (parameters.D == null)
+            {
+                includePrivate = false;
```

Comment:
Does this really need to be passed to the two helper methods? They could check D for null just as easily as checking this bool.